### ISSUE TITLE
Add support for rsample to MixtureSameFamily Distribution

### ISSUE THREAD

**Issue by malbergo** [pytorch/pytorch#39178]
## 🚀 Feature

It would be great if MixtureSameFamily had the rsample() functionality so as to make the parameters of the MixturesameFamily distribution trainable. 

## Motivation

rsample() is a valuable tool for making the parameters defining distributions trainable. Mixture Models have recently been added to the set of distributions pytorch provides, but the reparameterization trick cannot be applied to them, for, say, a mixture of gaussians.

## Pitch

It would be great if this could be implemented! Not much of a pitch to sell here, but just think having such functionality if possible would be a great addition to what MixtureSameFamily can offer. 

## Alternatives

One could feasibly do this manually by hardcoding the mixture distribution, but I think that defeats the purpose of having the MixtureSameFamily option

## Additional context

Thanks for your help! Totally understand if for some  reason this isn't implementable or low on priority list.

**Comment by muupan** [comment]
👍 for this feature. TensorFlow's `MixtureSameFaimily` has a `reparameterize` option using using implicit reparameterization gradients. https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/MixtureSameFamily

**Comment by malbergo** [comment]
Just wanted to see if there was any update on this front :)

**Comment by jackshi0912** [comment]
Is there any update for this? I think differentiating through GMM RV is a fairly common use-case. Right now the lack of this feature makes modelling contiguous multimodal distributions a pain.

**Comment by fritzo** [comment]
Another approach might be @martinjankowiak's [GaussianScaleMixture](https://docs.pyro.ai/en/stable/distributions.html#pyro.distributions.GaussianScaleMixture) distribution in Pyro, but it may be more limited than TFP's implicit reparametrization

**Comment by zxc8594** [comment]
I truly hope so

**Comment by malbergo** [comment]
Any update on this front?

**Comment by Liam-Tian** [comment]
Is there any update on this one?

**Comment by CXX1113** [comment]
Is there any update on this one?

**Comment by haochuan-li** [comment]
Is there any update on this one?

**Comment by panda2020-sky** [comment]
Is there any update on this one?

**Comment by robsyc** [comment]
Does anyone know a solid way to learn attentions value for multiple distributions and subsequently combining them aka MixtureSameFamily into a single one. Would be great to have this feature

**Comment by ankit204** [comment]
This will be a useful feature
